### DISCUSSION TITLE
Should we add `nvim-treesitter-context` for sticky header

### DISCUSSION THREAD

**Discussion by Saafo** [ayamir/nvimdots#693] [Ideas]
### Feature description

[nvim-treesitter-context](https://github.com/nvim-treesitter/nvim-treesitter-context) can implement the "sticky scroll" feature in VSCode, or the so called "sticky header" feature in Xcode 14, but I'm not sure whether every body like it.

**Comment by charliie-dev** [comment]
discussion would be a better place for this kind of topic.

**Comment by charliie-dev** [comment]
I tried this plugin before, but I removed it after few weeks.
It seems cool at first, but it's not very useful for me. I always tried to limit my functions or classes within the windows view of my terminal.

**Reply by Saafo** [reply]
Got it. I'm always dealing with large project and the length of most functions are very long, so I need it. I'll add to my custom config if it's not suitable to everybody.

**Reply by charliie-dev** [reply]
If that's the case, then indeed this plugin would be very helpful for u!

**Comment by ayamir** [comment]
I removed it about 1 year ago, but I forgot the reason about it: https://github.com/ayamir/nvimdots/commit/793db03

**Comment by vollowx** [comment]
It's really useful for developers who always edit big and indented files, so what about add it as a optional plugin (comment it) like `fcitx5.nvim` and `copilot.lua`?

**Reply by Saafo** [reply]
However, it seems `copilot.lua` isn't disabled by default?

https://github.com/ayamir/nvimdots/blob/3be4c9bc862ce8f67e4ce712cff893b96d215754/lua/modules/plugins/completion.lua#L57

**Reply by charliie-dev** [reply]
I have no opinion on adding a commented plugin. it depends on @ayamir 's decision.

**Comment by Saafo** [comment]
So since it's been removed long time ago, I'll add it as my custom plugin. Thanks everybody!

**Reply by ayamir** [reply]
It's ok to add it to `main` IMO. Feel free to open your pr for it.

**Reply by Saafo** [reply]
Ok, I've opened here #707 .
